Engine/Component/ComponentTransform2D.cpp

```diff
@@ -111,9 +115,47 @@ void ComponentTransform2D::GenerateGlobalModelMatrix()
 	}
 }
 
-void  ComponentTransform2D::SetSize(float width, float height)
+void  ComponentTransform2D::SetSize(float new_width, float new_height)
+{
+	width = new_width;
+	height = new_height;
+	OnTransformChange();
+}
+
+void ComponentTransform2D::SetPosition(float x, float y)
 {
-	rect.right = width;
-	rect.bottom = height;
+	position.x = x;
+	position.y = y;
 	OnTransformChange();
 }
+
+
+void ComponentTransform2D::SetPosition(float3* new_position)
+{
+	position = *new_position;
+	OnTransformChange();
+}
+
+void ComponentTransform2D::UpdateRect()
+{
+	rect.left = position.x - width / 2;
+	rect.right = position.x + width / 2;
+	rect.top = position.y - height / 2;
+	rect.bottom = position.y + height / 2;
+}
+
+void ComponentTransform2D::CalculateRectMatix(float new_width, float new_height, float4x4* matrix)
+{
+	CalculateRectMatix(rect.left, rect.top, new_width, new_height, matrix);
+	
+}
+
+void ComponentTransform2D::CalculateRectMatix(float x, float y, float new_width, float new_height, float4x4* matrix)
+{
+	*matrix = float4x4(global_matrix);
+	matrix->SetTranslatePart(float3(x, y, position.z));
+	*matrix = float4x4::RotateZ(rotation) * *matrix;
+	*matrix = *matrix * float4x4::RotateZ(-rotation);
+	*matrix = *matrix * float4x4::Scale(float3(new_width, new_height, 0));
```

Comment:
`CalculateRectMatix` has a typo in the name, also the parameter `matrix` should be passed as reference to avoid the referencing `*matrix` all the time. This is also a good practice because this way you ensure that the pointer direction is not modified.